lib/fizzy/execute.cpp

```diff
@@ -432,13 +432,14 @@ bool invoke_function(const FuncType& func_type, const F& func, Instance& instanc
     const auto num_args = func_type.inputs.size();
     assert(stack.size() >= num_args);
     span<const Value> call_args{stack.rend() - num_args, num_args};
-    stack.drop(num_args);
 
     const auto ret = func(instance, call_args, depth + 1);
     // Bubble up traps
     if (ret.trapped)
         return false;
 
+    stack.drop(num_args);
```

Comment:
If we want to be pedantic, shouldn't this happen before the trap?

Reply:
No. Trap terminates this execution, so we can keep stack unchanged.